runtime/binding-http/src/main/java/io/aklivity/zilla/runtime/binding/http/internal/config/HttpBindingConfig.java

```diff
@@ -252,173 +261,142 @@ public HttpRequestType resolveRequest(
+        HttpRequestType requestType,
+        String path)
     {
-        boolean isValid = true;
-        for (String name : request.pathParamValues.keySet())
+        Matcher matcher = requestType.pathMatcher.reset(path);
+        boolean matches = matcher.matches();
+        assert matches;
+
+        boolean valid = true;
+        for (String name : requestType.pathParams.keySet())
         {
-            Validator validator = request.pathParams.get(name);
-            if (validator != null)
+            String value = matcher.group(name);
+            if (value != null)
             {
-                String8FW value = request.pathParamValues.get(name);
-                if (!validator.read(value.value(), value.offset(), value.length()))
+                String8FW value0 = new String8FW(value);
+                Validator validator = requestType.pathParams.get(name);
+                if (!validator.read(value0.value(), value0.offset(), value0.length()))
                 {
-                    isValid = false;
+                    valid = false;
                     break;
                 }
             }
         }
-        return isValid;
+        return valid;
     }
 
     private boolean validateQueryParams(
-        HttpRequestType request)
+        HttpRequestType requestType,
+        String path)
     {
-        boolean isValid = true;
-        for (String name : request.queryParamValues.keySet())
+        Matcher matcher = requestType.queryMatcher.reset(path);
+        boolean valid = true;
+        boolean found = matcher.find();
+        while (found)
         {
-            Validator validator = request.queryParams.get(name);
+            String name = matcher.group(1);
+            Validator validator = requestType.queryParams.get(name);
             if (validator != null)
             {
-                String8FW value = request.queryParamValues.get(name);
+                String8FW value = new String8FW(matcher.group(2));
                 if (!validator.read(value.value(), value.offset(), value.length()))
                 {
-                    isValid = false;
+                    valid = false;
                     break;
                 }
             }
+            found = matcher.find();
         }
-        return isValid;
+        return valid;
     }
 
     public boolean validateContent(
-        HttpRequestType request,
+        HttpRequestType requestType,
         DirectBuffer buffer,
         int index,
         int length)
     {
-        boolean isValid = true;
-        if (request != null && request.content != null)
+        boolean valid = true;
+        if (requestType != null && requestType.content != null)
         {
-            Validator validator = request.content;
-            isValid = validator.read(buffer, index, length);
+            Validator validator = requestType.content;
+            valid = validator.read(buffer, index, length);
         }
-        return isValid;
+        return valid;
```

Comment:
```
return requestType != null &&
  requestType.content != null &&
  requestType.content.read(buffer, index, length);
```

Reply:
I don't think this code is identical to the previous. I went with
```
return requestType == null ||
            requestType.content == null ||
            requestType.content.read(buffer, index, length);
```
The intent is to accept things as valid if have no intention to validate them. Pls LMK.